pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -444,9 +447,30 @@ func getFirstContainerWithSourceVolume(containers []corev1.Container) (string, s
 
 // Delete deletes the component
 func (a Adapter) Delete(labels map[string]string) error {
-	if !utils.ComponentExists(a.Client, a.ComponentName) {
-		return errors.Errorf("the component %s doesn't exist on the cluster", a.ComponentName)
+	spinner := log.Spinner(fmt.Sprintf("Deleting devfile component %s", a.ComponentName))
+	defer spinner.End(false)
+
+	componentExists, err := utils.ComponentExists(a.Client, a.ComponentName)
+	if !componentExists || err != nil {
+		msg := fmt.Sprintf("Component %s does not exist", a.ComponentName)
+
+		// log the error if it failed to determine if the component exists
+		if err != nil {
+			msg = fmt.Sprintf("%s: %v", msg, err)
+		}
```

Comment:
We shouldn't say `Component %s does not exist` in case of an error asit might cause confusion.

Reply:
agreed here, also it looks complicated. Its better to simply return `err` here

Reply:
@mik-dass ok, i will try to just log the err then w/o saying component does not exist

@adisky Cant return the err here. This was asked by Tomas, we shouldn't return an err when no component exist:
1. if the kube user is `admin` and the namespace is deleted; the kube api tells there is no component present.
2. if the kube user is `developer` and the namespace is deleted; the kube api complains about not having sufficient RBAC policies to get deployments, and returns an error

So this was the reason why the err msg is being logged instead of being returned

Reply:
@maysunfaisal can we check the error code here? and wrap error accordingly?

Reply:
@adisky hmm let me see if i can clean it up further with `kerror`

Reply:
Updated

So I have now cleaned it up a bit more. I am specifically handling Forbidden errors on delete asides the Not Found errors as well. I have also put unit tests and made sure integration tests on CI for OpenShift and Minikube works with these specific errors!